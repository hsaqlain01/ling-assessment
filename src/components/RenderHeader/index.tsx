import React from 'react';
import {View, Text} from 'react-native';
import tw from 'twrnc';

const RenderHeader = () => {
  return (
    <View style={tw`flex-row w-full`}>
      <View
        style={tw`w-3/10 h-15 border-black items-center justify-center border-r-2 border-yellow-400  `}>
        <Text style={tw`my-3 text-black mt-5`}>Name</Text>
      </View>
      <View
        style={tw`w-2/10 h-15 border-black border-r-2   border-yellow-400  items-center justify-center`}>
        <Text style={tw`my-3 text-black  mt-7`}>Rank</Text>
      </View>
      <View
        style={tw`w-2/10  h-15 border-black border-r-2 items-center justify-center  border-yellow-400 `}>
        <Text style={tw`my-3 text-Black  mt-7`}>Bananas</Text>
      </View>
      <View
        style={tw`w-3/10 h-15 border-white items-center justify-center px-1 `}>
        <Text style={tw`my-3 text-black text-center text-xs mt-7`}>
          isSearchedUser?
        </Text>
      </View>
    </View>
  );
};

export default RenderHeader;
