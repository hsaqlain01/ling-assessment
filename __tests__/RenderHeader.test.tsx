import React from 'react';
import renderer from 'react-test-renderer';
import RenderHeader from '../src/components/RenderHeader';

it('Header renders correctly', () => {
  const tree = renderer.create(<RenderHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});
