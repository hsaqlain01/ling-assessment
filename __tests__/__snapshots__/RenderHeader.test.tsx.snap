// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#facc15",
        "borderRightWidth": 2,
        "height": 60,
        "justifyContent": "center",
        "width": "30%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "marginBottom": 12,
          "marginTop": 20,
        }
      }
    >
      Name
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#facc15",
        "borderRightWidth": 2,
        "height": 60,
        "justifyContent": "center",
        "width": "20%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "marginBottom": 12,
          "marginTop": 28,
        }
      }
    >
      Rank
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#facc15",
        "borderRightWidth": 2,
        "height": 60,
        "justifyContent": "center",
        "width": "20%",
      }
    }
  >
    <Text
      style={
        {
          "marginBottom": 12,
          "marginTop": 28,
        }
      }
    >
      Bananas
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#fff",
        "height": 60,
        "justifyContent": "center",
        "paddingLeft": 4,
        "paddingRight": 4,
        "width": "30%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 12,
          "lineHeight": 16,
          "marginBottom": 12,
          "marginTop": 28,
          "textAlign": "center",
        }
      }
    >
      isSearchedUser?
    </Text>
  </View>
</View>
`;
